lang/rust/patches/0002-rustc-bootstrap-cache.patch

```diff
@@ -41,3 +41,15 @@
          let cache_dir = cache_dst.join(key);
          if !cache_dir.exists() {
              t!(fs::create_dir_all(&cache_dir));
+@@ -704,7 +710,10 @@ download-rustc = false
+         let llvm_assertions = self.llvm_assertions;
+ 
+         let cache_prefix = format!("llvm-{llvm_sha}-{llvm_assertions}");
+-        let cache_dst = self.out.join("cache");
++        let cache_dst = match env::var_os("OPENWRT_RUSTC_BOOTSTRAP_CACHE") {
```

Comment:
Probably it is better to upstream the feature with a generic name so it won't be a problem in the future...

I'll try to do that this evening if nobody beats me at it :)

Reply:
@jefferyto already created a PR for it: https://github.com/rust-lang/rust/pull/116697
But I think he is busy these months...

Reply:
Thank you for reminding me, let me see if I can bake a newer one based on the review comment.

Reply:
I rebased and addressed a suggestion https://github.com/rust-lang/rust/pull/121976

Reply:
It landed https://github.com/rust-lang/rust/pull/121976 shall we use it now or wait for the next release?

Reply:
> shall we use it now

Sounds great ;)